src/sinks/datadog/traces/stats.rs

```diff
@@ -0,0 +1,532 @@
+            None => {
+                let mut gs = GroupedStats::new();
+                Bucket::update(span, weight, is_top, &mut gs);
+                self.data.insert(aggkey, gs);
+            }
+        }
+    }
+
+    fn update(span: &BTreeMap<String, Value>, weight: f64, is_top: bool, gs: &mut GroupedStats) {
+        is_top.then(|| {
+            gs.top_level_hits += weight;
+        });
+        gs.hits += weight;
+        let error = match span.get("error") {
+            Some(Value::Integer(val)) => *val,
+            _ => 0,
+        };
+        if error != 0 {
+            gs.errors += weight;
+        }
+        let duration = match span.get("duration") {
+            Some(Value::Integer(val)) => *val,
+            _ => 0,
+        };
+        gs.duration += (duration as f64) * weight;
+        if error != 0 {
+            gs.err_distribution.insert(duration as f64)
+        } else {
+            gs.ok_distribution.insert(duration as f64)
+        }
+    }
+}
+
+struct Aggregator {
+    buckets: BTreeMap<u64, Bucket>,
+}
+
+impl Aggregator {
+    fn new() -> Self {
+        Self {
+            buckets: BTreeMap::new(),
+        }
+    }
+
+    fn handle_trace(&mut self, partition_key: &PartitionKey, trace: &TraceEvent) {
+        let spans = match trace.get("spans") {
+            Some(Value::Array(v)) => v.iter().filter_map(|s| s.as_object()).collect(),
+            _ => vec![],
+        };
+
+        let weigth = extract_weigth_from_root_span(&spans);
+        let payload_aggkey = PayloadAggregationKey {
+            env: partition_key.env.clone().unwrap_or_default(),
+            hostname: partition_key.hostname.clone().unwrap_or_default(),
+            version: trace
+                .get("app_version")
+                .map(|v| v.to_string_lossy())
+                .unwrap_or_default(),
+            container_id: trace
+                .get("container_id")
+                .map(|v| v.to_string_lossy())
+                .unwrap_or_default(),
+        };
+
+        spans.iter().for_each(|span| {
+            let is_top = metric_flag(span, TOP_LEVEL_KEY);
+            if !(is_top
+                || metric_flag(span, MEASURED_KEY)
+                || metric_flag(span, PARTIAL_VERSION_KEY))
+            {
+                return;
+            }
+            self.handle_span(span, weigth, is_top, payload_aggkey.clone());
+        });
+    }
+
+    fn handle_span(
+        &mut self,
+        span: &BTreeMap<String, Value>,
+        weight: f64,
+        is_top: bool,
+        payload_aggkey: PayloadAggregationKey,
+    ) {
+        let aggkey = AggregationKey::new_aggregation_from_span(span, payload_aggkey);
+        let start = match span.get("start") {
+            Some(Value::Timestamp(val)) => val.timestamp_nanos(),
+            _ => Utc::now().timestamp_nanos(),
+        };
+        // 10 seconds bucket
+        let btime = (start - start % 10_000_000_000) as u64;
```

Comment:
```suggestion
        let btime = (start - (start % 10_000_000_000)) as u64;
```

To leave no room for ambiguity here without needing to pull out the operator precendence table 😄